src/src/com/microsoft/aad/adal/BlobContainer.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+class BlobContainer {
+
+    /**
+     * universal version identifier for BlobContainer class
+     */
+    private static final long serialVersionUID = 5722517776407773544L;
+    
+    private static final String TAG = "BlobContainer";    
+
+    /**
+     * the BlobContainer stores the FRT tokenCacheItem 
+     * of the given user
+     */
+    private TokenCacheItem tokenItem;
+    
+    /**
+     * Check if the cache item contains family refresh token
+     * if true, value the tokenItem with this cache item
+     * @param item
+     */
+    BlobContainer(TokenCacheItem tokenItem) {
+        //only FRT is stored here 
+        if(tokenItem == null || StringExtensions.IsNullOrBlank(tokenItem.getFamilyClientId())) {
+            if(tokenItem != null) {
```

Comment:
I am wondering if we should to treat the illegal input of the constructor differently. 
1. if the input arg is null, it will throw the IllegalArgumentException
2. if the input tokenItem is not a family refresh token item, what should we do? 
    - log the error
    - throw IllegalArgumentException
    - throw the authentication exception